Short lived container - docker socket update (#5049)
DataDog/documentation: content/en/agent/kubernetes/daemonset_setup.md

{{% /tab %}}
{{% tab "Docker Socket" %}}

By default the Agent receive updates on container in real time through Docker Events. However the configuration that is extracted from the container labels (autodiscovery) is updated every 10 seconds by the Agent. So any container with a shorter duration life might not have any data collected by the Agent.

> You can override this interval with a shorter one by setting the `ad_config_poll_interval` parameter which correspond to the `DD_AD_CONFIG_POLL_INTERVAL` environment variable.

Reviewer: ```suggestion
You can override this interval with a shorter one by setting the `ad_config_poll_interval` parameter, which is equivalent to the `DD_AD_CONFIG_POLL_INTERVAL` environment variable.
```

> The expected value is a integer in seconds or you can specify the unit, for example:

Reviewer 2: actually the example below is not the same parameter, `DD_AD_CONFIG_POLL_INTERVAL` is the global setting for all config providers (in seconds), below is the setting for an individual config provider unfortunately you cannot use an env var for this setting as it's within a list

Author: So do you suggest that we recommend to update the global one or just the one from the docker provider?

Reviewer 2: on a docker setup the easier is to go with `DD_AD_CONFIG_POLL_INTERVAL` (but you cannot go bellow the second) on a docker+kubernetes setup (and if you use labels AD) I'd recommends using the indiviudal setting on docker to avoid polling the kubelet too often (we have a cache anyway)

Author: Let's focus on pure docker environment, in K8s we should recommend to go through files and pod annotations.

So I'll keep the `DD_AD_CONFIG_POLL_INTERVAL` and will change the example.

> The expected value is a integer in seconds or you can specify the unit as in the below example:

Reviewer: ```suggestion
The expected value is a integer in seconds or you can specify the unit, for example:
```

{{% /tab %}}
{{% tab "Docker Socket" %}}

> For Docker environment (no Kubernetes), the Agent receives container updates in real time through Docker events. However, the configuration that is extracted from the container labels (Autodiscovery) is updated every 10 seconds by the Agent. Therefore, any container with a shorter life may not have any data collected by the Agent.

Reviewer: ```suggestion
For Docker environment (not Kubernetes), the Agent receives container updates in real time through Docker events. However, the configuration that is extracted from the container labels (Autodiscovery) is updated every 10 seconds by the Agent. Therefore, any container with a shorter life may not have any data collected by the Agent.
```

> For Docker environment (not Kubernetes), the Agent receives container updates in real time through Docker events. However, the configuration that is extracted from the container labels (Autodiscovery) is updated every 10 seconds by the Agent. Therefore, any container with a shorter life may not have any data collected by the Agent.

Reviewer: ```suggestion
For a Docker environment, the Agent receives container updates in real time through Docker events. The Agent extracts and updates the configuration from the container labels (Autodiscovery) every 10 seconds. Therefore, by default, the Agent does not collect data for any container with a shorter life.
```

> You can override this interval with a shorter one by setting the `ad_config_poll_interval` parameter, which is equivalent to the `DD_AD_CONFIG_POLL_INTERVAL` environment variable.

Reviewer: ```suggestion
You can override the polling interval by setting the `ad_config_poll_interval` parameter, which is equivalent to the `DD_AD_CONFIG_POLL_INTERVAL` environment variable.
```

> For a Kubernetes environment, use the `K8s file` collection method that supports init, stopped, and short lived containers collection without any extra setup.

Reviewer: ```suggestion
For a Kubernetes environment, use the `K8s file` collection method that supports init, stopped, and short lived containers. No additional setup is required.
```

> In Kubernetes environment, we recommend to use the `K8s file` collection method that supports init, stopped, and short lived containers collection without any extra setup.

Reviewer: ```suggestion
For a Kubernetes environment, use the `K8s file` collection method that supports init, stopped, and short lived containers collection without any extra setup.
```
